### ISSUE TITLE
error: not supported: mentix: provider mesh.pondersource.com has no supported services

### ISSUE THREAD

**Issue by michielbdejong** [pondersource/sciencemesh-php#122]
This error is logged in the Reva logs when I try to create an invite. It happens both on the pondersource.org and the pondersource.com site. I don't know if it's a bug in the Mentix driver for OCM provider authorizer, or whether gocdb is not listing services in the place where the mentix driver expects them to be.

I'll debug this.

**Comment by michielbdejong** [comment]
This also happens when trying to accept an invite, by the way.

**Comment by michielbdejong** [comment]
Debugging this now on
cloud.pondersource.org ->
mesh.pondersource.org ->
mesh.pondersource.com ->
cloud.pondersource.com

**Comment by michielbdejong** [comment]
Hm, with that, the error changed to:
```
2023-01-03 15:07:28.968 ERR ../../reva/internal/http/interceptors/providerauthorizer/providerauthorizer.go:103 > provider not registered in OCM error="error: not supported: No IP provided" pid=143 pkg=rhttp traceid=c8acdccf551d9d90e146c2703781aa76
```

**Comment by michielbdejong** [comment]
Hm, it looks from the code that
https://github.com/cs3org/reva/blob/v1.22.0/internal/http/interceptors/providerauthorizer/providerauthorizer.go#L97-L99

always calls IsProviderAllowed with a struct that only has a Domain and no Services and so https://github.com/cs3org/reva/blob/v1.22.0/pkg/ocm/provider/authorizer/mentix/mentix.go#L192 will always error.

The fix would probably be in https://github.com/cs3org/reva/blob/v1.22.0/pkg/ocm/provider/authorizer/mentix/mentix.go#L179 to store the matching provider, and then look there.

**Comment by michielbdejong** [comment]
I wonder how this has ever worked. Maybe it has just always been broken and I had just always misconfigured the http interceptor for the provider authorizer until now?

**Comment by michielbdejong** [comment]
Hm, it is still checking `127.0.1.1 ?= mesh.pondersource.org` so I decided to set `verify_request_hostname` to false, see https://github.com/cs3org/ocm-test-suite/commit/aa600adbf89e85f4bf73c955d4cddd4c20f32b8e

But this is obviously not satisfactory, will bring this up in https://gitter.im/sciencemesh/task-force-technical .

**Comment by michielbdejong** [comment]
I published a Docker image [michielbdejong/reva:mentix-fixes](https://hub.docker.com/r/michielbdejong/reva/tags) so kubernetes users can test with that.

**Comment by mirekys** [comment]
@michielbdejong I'll try to have a look into whats going on there

**Comment by mirekys** [comment]
> I wonder how this has ever worked. Maybe it has just always been broken and I had just always misconfigured the http interceptor for the provider authorizer until now?

I think that that http interceptor was not really being used up until now with NC/OC integrations, everything went through the grpc implementation here: https://github.com/cs3org/reva/blob/master/internal/grpc/services/ocmproviderauthorizer/ocmproviderauthorizer.go#L120

**Comment by mirekys** [comment]
@michielbdejong to what IP address does the `mesh.pondersource.org` resolve to from inside your container? What error message does that exactly produce?

**Comment by michielbdejong** [comment]
Ah, good question! Indeed, that seems to have been part of the problem, I had apparently set it to 127.0.1.1 in /etc/hosts - fixed that now. But still seeing:
```
2023-01-05 16:02:14.484 ERR ../reva/internal/http/interceptors/providerauthorizer/providerauthorizer.go:101 > provider not registered in OCM error="error: bad request: Invalid requesting OCM endpoint IP mesh.pondersource.org of provider mesh.pondersource.org" pid=503475 pkg=rhttp traceid=ba4b6d299907126c3e7fcecbe649f5b2
```
when trying to run with `verify_request_hostname = true`. Is this working on your site?

**Comment by michielbdejong** [comment]
> everything went through the grpc implementation

Right, I can confirm that ocm-invite-generate works from reva-cli but not if it's coming in over http.

What is this feature supposed to do, though? IWhen I use reva-cli from my laptop it's also allowing the request even though I would expect my laptop to have an invalid requesting OCM endpoint IP?

**Comment by michielbdejong** [comment]
Where is the documented? I only found https://github.com/cs3org/reva/pull/883 about it.

**Comment by labkode** [comment]
@michielbdejong If I recall correctly, this check is used to avoid receiving a OCM share request from sites not part of the mesh (meaning sites not listed in Mentix).

**Comment by michielbdejong** [comment]
This is resolved in latest Reva because the REST points that the EFSS calls are now on /sciencemesh instead of on /ocm.

**Comment by michielbdejong** [comment]
I'll test what happens if we remove https://github.com/cs3org/ocm-test-suite/blob/main/servers/revad/sciencemesh.toml#L75

**Comment by michielbdejong** [comment]
That seemed to work fine on first test! Removing it.
